src/util-hugepages.c

```diff
@@ -38,15 +38,15 @@ static void SystemHugepageSnapshotDump(SystemHugepageSnapshot *s);
 
 static bool SystemHugepageSupported(void)
 {
-#if !defined __CYGWIN__ && !defined OS_WIN32 && !defined __OpenBSD__ && !defined sun
+#if defined __linux__
```

Comment:
what was the reason to make this use ifdefs instead of just checking at runtime if hugepage support is enabled?

Reply:
the function seems wrong, because it doesnt seem to be true that linux means hugepages are enabled

Reply:
The guards were there to avoid Windows compilation errors: 
https://github.com/lukashino/suricata/actions/runs/7004361732/job/19052076585

extracted from: https://github.com/OISF/suricata/pull/9898

Reply:
that seems like a pretty small issue that can be addressed by refactoring things a bit, and/or investigating how to do this properly on windows

Reply:
also, it seems it should work, so perhaps it's an include issue? https://github.com/luzexi/MinGW/blob/master/x86/include/dirent.h#L26